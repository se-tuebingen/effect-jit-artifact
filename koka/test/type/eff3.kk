fun choose(x,y){ if (True) then x else y}

fun test() {
  choose( ref(1), throw("hi") )
}

fun test2() {
  choose( const(())(ref(1)), ref(2).set(3))
}
