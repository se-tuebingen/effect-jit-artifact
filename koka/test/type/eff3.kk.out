type/eff3/choose: forall<a> (x : a, y : a) -> a
type/eff3/test: forall<h> () -> <alloc<h>,exn> ref<h,int>
type/eff3/test2: () -> ()